src/problems/sudoku_Leuni.clj

```diff
@@ -0,0 +1,192 @@
+        y-pos (second(:current-pos state))]
+
+    ;For numbers 1 through 9, x, create a new state with the number at current-pos changed to the new number x + 1
+    ;If current-pos is non-zero create children that have the same board and incremented current-pos
+    (for [x (range 9)] (if (> (get (get board x-pos) y-pos) 0)
+                         (State. board (if (= x-pos 8) [0 (+ 1 y-pos)] [(+ 1 x-pos) y-pos] ))
+
+                         ;Else change the number at curret-pos to x+1
+                         (State. (assoc board x-pos (assoc (get board x-pos)  y-pos (+ 1 x)))
+
+                                 ;If current-pos is [8 8] don't increment, return the same current-pos with the new board
+                                 (if (and (= x-pos 8) (= y-pos 8)) [8 8]
+
+                                   ;Else if current-pos x-pos is 8 set current-pos to [0 (y-pos + 1)
+                                   (if (= x-pos 8) [0 (+ 1 y-pos)] [(+ 1 x-pos) y-pos])))))))
+
+
+; Function   : check-square
+;
+; Description: Given a state [board current-pos] this
+;              function will determine if there are any duplicate values
+;              in the current-pos ([row, col]) square.
+;
+; Returns    : true  - if no duplicates (all elements are distinct)
+;              false - if duplicates (all elements are not distinct)
+(defn check-square[state pos]
+  (let [board (:board state)
+        cur-row (first pos)
+        cur-col (second pos)]
+
+      ; Determine the appropriate 'square column(legal? start-state)'
+      ; by using which column is the current column
+      (def col-square (if (<= cur-col 2) 0
+                        (if (>= cur-col 6) 2
+                          1)))
+
+      ; Determine the appropriate 'square row'
+      ; by using which column is the current column
+      (def row-square (if (<= cur-row 2) 0
+                        (if (>= cur-row 6) 2
+                          1)))
+
+      ; Determine the index of the upper-left
+      ; corner of the current square
+      (def row-square-corner (* row-square 3))
+      (def col-square-corner (* col-square 3))
+
+      ; Pull each row of the board that contains
+      ; the current square
+      (def first-row (get board row-square-corner))
+
+      (def second-row (get board (+ row-square-corner 1)))
+
+      (def third-row (get board (+ row-square-corner 2)))
+
+      ; Partition those rows to get the square
+      (def first-row-partition (subvec first-row col-square-corner (+ col-square-corner 3)))
+
+      (def second-row-partition (subvec second-row col-square-corner (+ col-square-corner 3)))
+
+      (def third-row-partition (subvec third-row col-square-corner (+ col-square-corner 3)))
+
+      ; Build the square from the partitions
+      (def square (vec (concat first-row-partition second-row-partition third-row-partition)))
+
+      ; Determine if the square has no non-zero duplicates
+      (apply distinct? (filter #(> % 0) square))
```

Comment:
You end up repeating this logic quite a few times – a new function would be good.